lib/ljsyscall/syscall/linux/syscalls.lua

```diff
@@ -457,9 +457,28 @@ function S.sched_setaffinity(pid, mask, len) -- note len last as rarely used
   return retbool(C.sched_setaffinity(pid or 0, len or s.cpu_set, mktype(t.cpu_set, mask)))
 end
 
+local function get_max_numnodes()
+  local name = "/proc/self/status"
+  local fd, err = S.open(name, c.O.RDONLY)
```

Comment:
Here we would need to return a default value, but then below we are only reading 8192 bytes, whereas for 1024 nodes that would already take 1024*9=9216 bytes.  So this won't scale arbitrarily.  I think it could be OK to instead of calling get_max_numnodes, just always use 4096 or something large.  Avoids many potential problems.  WDYT?

Reply:
OK with me if it will always work.